assets/js/panel/panels/WorkflowRunPanel.tsx

```diff
@@ -0,0 +1,80 @@
+import { ManualRunPanel } from "../../manual-run-panel/ManualRunPanel"
+import type { WithActionProps } from "#/react/lib/with-props"
+import { Panel } from "../Panel"
+import React from "react"
+
+
+interface ManualRunBody {
+  manual: {
+    body: string | null
+    dataclip_id: string | null
+  }
+}
+
+interface WorkflowRunPanel {
+  job_id: string
+  cancel_url: string
+}
+
+export const WorkflowRunPanel: WithActionProps<WorkflowRunPanel> = (props) => {
+  const { job_id, cancel_url, pushEvent, ...actionProps } = props;
+  const [manualContent, setManualRunContent] = React.useState<ManualRunBody>({ manual: { body: null, dataclip_id: null } })
+
+  const runDisabled = React.useMemo(() => {
```

Comment:
I'm sure this can be simplified a lot? If we had a unit test framework here I'd push this a bit harder.

It looks like an important function and it's very hard for me to see the rules of when a run is disabled